py/farm_ng/core/event_service_recorder.py

```diff
@@ -95,6 +101,9 @@ def __init__(self, service_name: str, config_list: EventServiceConfigList) -> No
         self.record_queue: asyncio.Queue[tuple[event_pb2.Event, bytes]] = asyncio.Queue(
             maxsize=self.QUEUE_MAX_SIZE,
         )
+        self.header_deque: deque[tuple[event_pb2.Event, bytes]] = deque()
+        if header_msgs:
```

Comment:
```suggestion
        if header_msgs is not None:
```
to avoid someone passing a bool or string and append that. Maybe we should check on the type of the header message bedfore appending to the queue

Reply:
Yeah good call. We check in `EventsFIleWriter`, but better to check further upstream